test/api/buffer/api_test.go

```diff
@@ -0,0 +1,398 @@
+//nolint:forbidigo
+package api
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/keboola/go-utils/pkg/orderedmap"
+	"github.com/keboola/go-utils/pkg/wildcards"
+	"github.com/spf13/cast"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/encoding/json"
+	"github.com/keboola/keboola-as-code/internal/pkg/env"
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem/aferofs"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/testhelper"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/testhelper/storageenv"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/testproject"
+	"github.com/keboola/keboola-as-code/internal/pkg/validator"
+)
+
+// TestBufferApiE2E runs one functional test per each subdirectory.
+func TestBufferApiE2E(t *testing.T) {
```

Comment:
Dala by sa nejak ta logika zobvseobecnit a dat do samostatneho pkg?
IMHO sa to cele lisi iba v binarke, ktora sa spusta.

Reply:
V tomto pripade je to nejaky interny nastroj na testovanie API, neocakava sa, ze sa bude medzi API lisit.

Reply:
Asi dala, ale nebude to úplně jednoduchý, protože templates používá repositories, takže se tam musí dostat ten option při startupu. Možná toho bude ještě víc. Chtěl bych to ale řešit mimo, už tak se při tom setupu nějaký věci duplikovaly, tak jsem na to udělal další task v cooldownu https://keboola.atlassian.net/browse/BA-76

Reply:
ok, moze byt